src/shop-payment-request.html

```diff
@@ -0,0 +1,283 @@
+          var newShippingOption, oldShippingOption;
+          details.shippingOptions.forEach( function(shippingOption) {
+              if (shippingOption.id === shippingOptionId) {
+                  newShippingOption = shippingOption;
+              } if (shippingOption.selected === true) {
+                  oldShippingOption = shippingOption;
+              }
+          });
+          // Select the new, deselect the old
+          oldShippingOption.selected = false; // Do this first, in case the same one is selected
+          newShippingOption.selected = true;
+          // Update the total cost and line items
+          this._updateTotal(details, newShippingOption, oldShippingOption);
+
+          resolve(details);
+      },
+
+      // Updates the total order cost to include shipping
+      // If shipping has already been added, it must be subtracted and the line item removed
+      _updateTotal: function(details, newShippingOption, oldShippingOption) {
+          var oldShippingCost = 0;
+          if (oldShippingOption) {
+              // Remove from the line items; it should be on the end
+              // TODO: explicitly remove the item by its id?
+              details.displayItems.pop();
+
+              // Deduct the cost of the old shipping
+              oldShippingCost = Number(oldShippingOption.amount.value);
+          }
+
+          var shippingCost = Number(newShippingOption.amount.value);
+          var currentTotal = Number(details.total.amount.value);
+          details.total.amount.value = (currentTotal + shippingCost - oldShippingCost).toFixed(2);
+          // Place the shipping price at the end of the line items
+          details.displayItems.splice(details.displayItems.length, 0, newShippingOption);
+      },
+
+      _computePaymentTypesSupported: function(supportedMethods, androidPay) {
+        var paymentTypes = [];
+
+        if (supportedMethods.length > 0) {
+          paymentTypes.push({
+            supportedMethods: supportedMethods
+          });
+        }
+
+        if (androidPay && androidPay != {}) {
+          paymentTypes.push({
+            supportedMethods: ['https://android.com/pay'],
+            data: androidPay
+          });
+        }
+
+        return paymentTypes;
+      },
+
+      _computeItemDetails: function() {
+        var items = [];
+
+        if (this.cart) {
+          for (var i = 0; i < this.cart.length; ++i) {
+            var entry = this.cart[i];
+            var itemCost = entry.quantity * entry.item.price;
+            items.push({
+              label: entry.item.title + ' ' + entry.size + ' x' + entry.quantity,
+              amount: {currency: this.currency, value: itemCost.toFixed(2)}
+            });
+          }
+        }
+
+        return items;
+      },
+
+      _redirectToCheckout: function() {
+        // prevent reload loop if we're already on /checkout.
+        if (this.page !== 'checkout') {
+          location.href = '/checkout';
```

Comment:
The redirection seems to be reloading the app.  If the redirection were handled in `shop-app`, it would be possible to get a reference to the `app-location` and change the route: e.g. `this.$.location.set('route.path', '/');` wdyt?